mlir/lib/Dialect/Utils/ReshapeOpsUtils.cpp

```diff
@@ -451,7 +451,9 @@ mlir::getSymbolLessAffineMaps(ArrayRef<ReassociationExprs> reassociation) {
   SmallVector<AffineMap, 4> maps;
   maps.reserve(reassociation.size());
   for (const auto &exprs : reassociation) {
-    assert(!exprs.empty());
+    if (exprs.empty()) {
+      return {};
+    }
```

Comment:
Generally, compiler code should assume/assert that the IR has passed the verifier, so this should stay as an assertion unless it is somehow called on unverified code.